sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -209,14 +210,19 @@ private boolean shouldApplyScopeData(
       sentryId = SentryId.EMPTY_ID;
     }
 
-    // if we encountered an abnormal exit finish tracing in order to persist and send
+    // if we encountered a crash/abnormal exit finish tracing in order to persist and send
     // any running transaction / profiling data
     if (scope != null) {
-      @Nullable ITransaction transaction = scope.getTransaction();
+      final @Nullable ITransaction transaction = scope.getTransaction();
       if (transaction != null) {
-        // TODO if we want to do the same for crashes, e.g. check for event.isCrashed()
         if (HintUtils.hasType(hint, TransactionEnd.class)) {
```

Comment:
I'm assuming there's no risk of this firing multiple times, leading to multiple forceFinish invocations where the root span hasn't yet been marked as finished before consecutive invocations, correct?